dialect.go

```diff
@@ -8,40 +8,46 @@ import (
 // SqlDialect abstracts the details of specific SQL dialects
 // for goose's few SQL specific statements
 type SqlDialect interface {
-	createVersionTableSql() string // sql string to create the goose_db_version table
-	insertVersionSql() string      // sql string to insert the initial version table row
-	dbVersionQuery(db *sql.DB) (*sql.Rows, error)
+	createVersionTableSql(name string) string // sql string to create the goose_db_version table
+	insertVersionSql(name string) string      // sql string to insert the initial version table row
+	dbVersionQuery(db *sql.DB, name string) (*sql.Rows, error)
 }
 
-var dialect SqlDialect = &PostgresDialect{}
+func (c *Client) GetDialect() SqlDialect {
```

Comment:
Agreed. We need the `func GetDialect() SqlDialect` to retain API compatibility, but this method should be unnecessary.